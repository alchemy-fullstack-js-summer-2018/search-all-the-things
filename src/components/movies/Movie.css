:local(.movie) {
  padding: 5px 10px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 10px;
  text-align: center;
  background: #eee;
  width: 250px;
  height: 350px;
  border-radius: 5px;

  img {
    width: 70%;
    height: auto;
  }
}