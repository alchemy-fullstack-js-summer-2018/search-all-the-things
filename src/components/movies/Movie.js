import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './Movie.css';

export default class Movie extends Component {

    static propTypes = {
      movie: PropTypes.object.isRequired
    };

    render() {
      const { movie } = this.props;
      const { Title, Year, Poster, imdbID } = movie;

      return (
        <li className={styles.movie}>
          <Link to={`/movies/${imdbID}`}>
            {Poster !== 'N/A'
              ? <img src={Poster}/>
              : <img src='https://www.nationalpetregister.org/assets/img/no-photo.jpg'/>}
            <p><strong>{Title}</strong> ({Year})</p> 
          </Link>
        </li>
      );
    }
}