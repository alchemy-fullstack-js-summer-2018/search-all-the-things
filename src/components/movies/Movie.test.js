import * as React from 'react';
import { shallow, configure  } from 'enzyme';
import Movie from './Movie';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const movie = 
  {  'Title':'Blade Runner', 'Year':'1982', 'Rated':'R', 'Released':'25 Jun 1982', 'Runtime':'117 min', 'Genre':'Sci-Fi, Thriller', 'Director':'Ridley Scott', 'Writer':'Hampton Fancher (screenplay), David Webb Peoples (screenplay), Philip K. Dick (novel)', 'Actors':'Harrison Ford, Rutger Hauer, Sean Young, Edward James Olmos', 'Plot':'A blade runner must pursue and terminate four replicants who stole a ship in space, and have returned to Earth to find their creator.', 'Language':'English, German, Cantonese, Japanese, Hungarian, Arabic', 'Country':'USA, Hong Kong', 'Awards':'Nominated for 2 Oscars. Another 11 wins & 16 nominations.', 'Poster':'https://m.media-amazon.com/images/M/MV5BNzQzMzJhZTEtOWM4NS00MTdhLTg0YjgtMjM4MDRkZjUwZDBlXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SX300.jpg', 'Ratings':[{  'Source':'Internet Movie Database', 'Value':'8.2/10'  }, {  'Source':'Rotten Tomatoes', 'Value':'90%'  }, {  'Source':'Metacritic', 'Value':'89/100'  }], 'Metascore':'89', 'imdbRating':'8.2', 'imdbVotes':'594,601', 'imdbID':'tt0083658', 'Type':'movie', 'DVD':'27 Aug 1997', 'BoxOffice':'N/A', 'Production':'Warner Bros. Pictures', 'Website':'N/A', 'Response':'True'  };

describe('Single Article test', () => {
    
  it('Renders as designed', () => {
    const wrapper = shallow(<Movie movie={movie}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Renders without image', () => {
    const noImageMovie = { ...movie, urlToImage: null };
    const wrapper = shallow(<Movie movie={noImageMovie}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});