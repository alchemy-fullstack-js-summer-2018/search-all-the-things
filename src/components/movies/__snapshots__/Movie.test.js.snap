// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Article test Renders as designed 1`] = `
<li
  className="movie"
>
  <Link
    replace={false}
    to="/movies/tt0083658"
  >
    <img
      src="https://m.media-amazon.com/images/M/MV5BNzQzMzJhZTEtOWM4NS00MTdhLTg0YjgtMjM4MDRkZjUwZDBlXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SX300.jpg"
    />
    <p>
      <strong>
        Blade Runner
      </strong>
       (
      1982
      )
    </p>
  </Link>
</li>
`;

exports[`Single Article test Renders without image 1`] = `
<li
  className="movie"
>
  <Link
    replace={false}
    to="/movies/tt0083658"
  >
    <img
      src="https://m.media-amazon.com/images/M/MV5BNzQzMzJhZTEtOWM4NS00MTdhLTg0YjgtMjM4MDRkZjUwZDBlXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SX300.jpg"
    />
    <p>
      <strong>
        Blade Runner
      </strong>
       (
      1982
      )
    </p>
  </Link>
</li>
`;
